airflow/operators/trigger_dagrun.py

```diff
@@ -128,6 +123,11 @@ def execute(self, context: Context):
         else:
             parsed_execution_date = timezone.utcnow()
 
+        try:
+            json.dumps(self.conf)
+        except TypeError:
+            raise AirflowException("conf parameter should be JSON Serializable")
+
```

Comment:
There was some discussion around allowing `XComArg`s in the initial issue, see [here](https://github.com/apache/airflow/issues/13414#issuecomment-776173348). One specific call out from @kaxil was:
> Currently, we don't support it as we need to de-serialize it for Webserver and show it in the UI in List DagRun page. Needs a bigger change to support other objects that are not JSON serializable atm

Can you do some testing around loading this page with your change @paolo-moriello to confirm that more code changes are required to support this.

Reply:
The change I'm proposing here is not changing the existing logic, apart from enabling the above code workflow which otherwise fails.

I've quickly tested that in the List DagRun UI I can see the both the triggered and the trigged dags, with the related config:
![Screenshot 2022-10-14 at 15 19 37](https://user-images.githubusercontent.com/61800102/195857000-ced71f8d-a4f6-4d54-aa5e-e19d2ca062cb.png)

Reply:
Thanks @paolo-moriello!
> I've quickly tested that in the List DagRun UI I can see the both the triggered and the trigged dags, with the related config:

And just to confirm, the config input your tested with here was an XComArg object?

Reply:
Yes, this is the code I've used:

```python
@dag
def trigger_pipeline(**kwargs):
    @task
    def foo() -> Dict[str, str]:
        return {"foo": "bar"}

    myconfig = foo() # (variable) myconfig: XComArg
    TriggerDagRunOperator(task_id='tdro', trigger_dag_id="listener_dag", conf=myconfig)
```